schematic/utils/schema_utils.py

```diff
@@ -90,19 +105,46 @@ def get_label_from_display_name(
         ValueError if entry_type.lower(), is not either 'class' or 'property'
 
     """
-    if entry_type.lower() == "class":
-        label = get_class_label_from_display_name(
-            display_name=display_name, strict_camel_case=strict_camel_case
-        )
-
-    elif entry_type.lower() == "property":
-        label = get_property_label_from_display_name(
-            display_name=display_name, strict_camel_case=strict_camel_case
+    if data_model_labels == "display_label":
+        blacklisted_chars = ["(", ")", ".", " ", "-"]
+        # Check that display name can be used as a label.
+        valid_display_name = check_if_display_name_is_valid_label(
+            display_name=display_name, blacklisted_chars=blacklisted_chars
         )
+        if valid_display_name:
+            label = display_name
+        else:
+            if entry_type.lower() == "class":
+                label = [
+                    get_class_label_from_display_name(str(display_name)).translate(
+                        {ord(x): "" for x in blacklisted_chars}
+                    )
+                ][0]
+
+            elif entry_type.lower() == "property":
+                label = [
+                    get_property_label_from_display_name(str(display_name)).translate(
+                        {ord(x): "" for x in blacklisted_chars}
+                    )
+                ][0]
+
+            logger.warning(
+                f"Cannot use display name {display_name} as the data model label, becaues it is not formatted properly. Please remove all spaces and blacklisted characters: {str(blacklisted_chars)}. The following label was assigned instead: {label}"
+            )
     else:
-        raise ValueError(
-            f"The entry type submitted: {entry_type}, is not one of the permitted types: 'class' or 'property'"
-        )
+        if entry_type.lower() == "class":
+            label = get_class_label_from_display_name(
+                display_name=display_name, strict_camel_case=strict_camel_case
+            )
+
+        elif entry_type.lower() == "property":
+            label = get_property_label_from_display_name(
+                display_name=display_name, strict_camel_case=strict_camel_case
+            )
+        else:
+            raise ValueError(
+                f"The entry type submitted: {entry_type}, is not one of the permitted types: 'class' or 'property'"
+            )
     return label
 
```

Comment:
This function is starting to look extremely complex. It shoudl probably be refactored into several smaller ones.

Reply:
I broke this function up, and added additional tests to cover the new functions :)